lib/datadog/core/configuration/option.rb

```diff
@@ -191,11 +189,33 @@ def coerce_env_variable(value)
           when :string, NilClass
             value
           else
-            raise ArgumentError,
+            raise InvalidDefinitionError,
               "The option #{@definition.name} is using an unsupported type option for env coercion `#{@definition.type}`"
           end
         end
 
+        def coerce_env_var_int(value)
+          # Strict integer parsing
+          Integer(value)
+        rescue ArgumentError => e
+          # Try also parsing as a whole floating point numbers
+          begin
+            f = Float(value)
+
+            # Check for whole numbers
+            raise ArgumentError, "#{value} is not a whole number" unless f.truncate == f
```

Comment:
To be honest... This seems like a bit of a surprising semantics. If I'm in a rush and see something on my config that I want to update:

```
DD_PROFILING_HOW_MUCH_IT_ROCKS=100.0
```

and then I go, oh, I want to make it rock a bit less

```
DD_PROFILING_HOW_MUCH_IT_ROCKS=95.5
```

and it blows up -- this seems weird. That .0 that someone on my team left misled me into thinking this value was supposed to be a float, and then it turns out it isn't and it doesn't work.

Reply:
What do you think about allowing whole floats as integer @TonyCTHsu @ekump? (e.g. `100.0` is allowed for `:int` configuration options). Or should be strict and say that this is not allowed (e.g. `100.0` would be rejected, and the user should replace it with an explicit `100`)

Reply:
@ivoanjo, how about a warning on whole floats? I think that might be a good compromise.

Reply:
> how about a warning on whole floats?

I would still kinda prefer not to accept 'em, but I can live with that as a compromise ;)

Reply:
We had discussed moving this to a warning, which I think is better than the current raise. If we do keep the raise, I would suggest making it a bit clear what needs to be fixed in the current error -- e.g. "Setting expects only integer settings, you've provided #{value}"?